app/schemas/devices.js

```diff
@@ -23,8 +23,42 @@ var update = {
   }
 };
 
+var create = {
+  thingd: joi
+    .object({
+      thingName: joi.string().required(),
+      modbusSlaveID: joi.number().min(0).max(255).required(),
+      modbusSlaveURL: joi.string().required(),
+      dataItems: joi.array().items(
+        joi.object({
+          schema: joi
+            .object({
+              sensorID: joi.number().required(),
+              sensorName: joi.string().required(),
+              typeID: joi.string().required(),
+              unit: joi.string().required(),
+              valueType: joi.string().required()
+            }),
+          modbus: joi
+            .object({
+              registerAddress: joi.number().min(0).max(65535).required(),
+              bitOffset: joi.number().only([1, 8, 16, 32, 64]).required(),
+            }),
+          config: joi
+            .object({
+              lowerThreshold: joi.number().required(),
```

Comment:
According to the [documentation](https://docs.google.com/document/d/1rWocy2MqWh_wkKmQWq9mGuDsUDgMgfV2ITERHGx2W-c/edit), these values won't necessarily always be passed. It depends on developer needs.

Reply:
Done!